pandas/core/dtypes/inference.py

```diff
@@ -28,20 +28,33 @@ def is_number(obj):
     """
     Check if the object is a number.
 
+    Booleans are valid because they are int subclass.
```

Comment:
did this come from a separate PR? Can you revert he changes here so they don't conflict with that PR?